tests/conftest.py

```diff
@@ -0,0 +1,511 @@
+                         "null": null_,
+                         "pval": z_pval(result_, null_)[1],
+                         "zval": z_pval(result_, null_)[0]
+                        }))
+
+        # make super sure the decoder is updated because I don't know if there's a reference guarantee
+        decoder.decoder = decoder_
+        decoder.trained = True
+
+
+""" 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Test Cases (Data)
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+"""
+
+
+# noinspection DuplicatedCode
+@DataRegistry.register()
+def base_data() -> Data:
+    """
+    Base dataset for testing.
+    """
+    data = generate_synthetic_data(n_neurons=80,
+                                   n_trials=200,
+                                   keyA="stimulus",
+                                   rateA=0.3,
+                                   keyB="action",
+                                   rateB=0.3,
+                                   timebins_per_trial=50,
+                                   corrAB=0.0,
+                                   scale=1.0,
+                                   meanfr=0.1,
+                                   mixed_term=0.0,
+                                   mixing_factor=0.0
+                                   )
+    return Data(
+        conditions=MappingProxyType({"stimulus": [-1, 1], "action": [-1, 1]}),
+        data=data,
+        key="base",
+        num_neurons=data.get("raster").shape[-1],
+        num_trials=len(set(data.get("trial").tolist())),
+        num_samples=data.get("raster").shape[0],
+        num_conditions=2,
+    )
+
+
+# noinspection DuplicatedCode
+@DataRegistry.register()
+def undersampled_data() -> Data:
+    """
+    An undersampled dataset for testing.
+    """
+    data = generate_synthetic_data(n_neurons=80,
+                                   n_trials=8,
+                                   keyA="stimulus",
+                                   rateA=0.3,
+                                   keyB="action",
+                                   rateB=0.3,
+                                   timebins_per_trial=1,
+                                   corrAB=0.0,
+                                   scale=1.0,
+                                   meanfr=0.1,
+                                   mixed_term=0.0,
+                                   mixing_factor=0.0
+                                   )
+    return Data(
+        conditions=MappingProxyType({"stimulus": [-1, 1], "action": [-1, 1]}),
+        data=data,
+        key="undersampled",
+        num_neurons=data.get("raster").shape[-1],
+        num_trials=len(set(data.get("trial").tolist())),
+        num_samples=data.get("raster").shape[0],
+        num_conditions=2,
+    )
+
+
+# noinspection DuplicatedCode
+@DataRegistry.register()
+def tangled_data() -> Data:
```

Comment:
I am not sure I get the "tangled" naming here. Might get confused with "entangled" which in the field means with non linear terms. Better to call it "correlated_variables" or something like that to be more explicit.

Reply:
This has been changed to `correlated_dataset`. "Tangled" was an exceptionally bad name, my bad! Not sure what I was thinking.